Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/ManipulationHandler.cs

```diff
@@ -0,0 +1,341 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using System.Linq;
+using UnityEngine.Assertions;
+using Microsoft.MixedReality.Toolkit.SDK.Input.Handlers;
+using Microsoft.MixedReality.Toolkit.Core.EventDatum.Input;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem.Handlers;
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Utilities;
+using Microsoft.MixedReality.Toolkit.Core.Services;
+using Microsoft.MixedReality.Toolkit.Core.Utilities.Physics;
+using System;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Utilities
+{
+    /// <summary>
+    /// This script allows for an object to be movable, scalable, and rotatable with one or two hands. 
+    /// You may also configure the script on only enable certain manipulations. The script works with 
+    /// both HoloLens' gesture input and immersive headset's motion controller input.
+    /// See Assets/HoloToolkit-Examples/Input/Readme/README_TwoHandManipulationTest.md
+    /// for instructions on how to use the script.
+    /// </summary>
+    /// 
+    public class ManipulationHandler : BaseFocusHandler, IMixedRealityInputHandler<MixedRealityPose>, IMixedRealitySourceStateHandler
+    {
+        #region Private Enums
+        private enum HandMovementType
+        {
+            oneHandedOnly = 0,
```

Comment:
I feel like some changes were reverted, like the capitalization changes here: https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/3323/commits/403140a93a374c8624088ee2a116d51270969607

Reply:
The enum capitalization might be the only one actually

Reply:
good catch!

Reply:
Updated!